lib/surface/compiler/eex_engine.ex

```diff
@@ -406,25 +406,39 @@ defmodule Surface.Compiler.EExEngine do
     ]
   end
 
-  defp handle_templates(
-         component,
-         [
-           %AST.SlotableComponent{
-             slot: name,
-             module: module,
-             let: let,
-             props: props,
-             templates: %{default: default}
-           }
-           | tail
-         ],
-         buffer,
-         state
-       ) do
+  defp handle_templates(component, [slotable | tail], buffer, state) do
+    %AST.SlotableComponent{
+      slot: name,
+      module: module,
+      let: let,
+      props: props,
+      templates: %{default: default}
```

Comment:
Why filter other slots? This prevents the user to use slots inside a slottable component.

The use case I have is the following:

```elixir
defmodule MyAppWeb.Layout do
  use MyAppWeb, :surface_component

  alias MyAppWeb.Layout.PageTitle
  alias MyAppWeb.Layout.Sidebar
  alias MyAppWeb.Layout.MobileHeader

  slot default, required: true
  slot page_title, as: :page_title_slot
  prop page_title, :string, required: false

  def render(assigns) do
    ~H"""
      <div
        class="h-screen flex overflow-hidden bg-white"
        x-data="{ sidebarOpen: false }"
        @keydown.window.escape="sidebarOpen = false"
      >
        <Sidebar />
        <div class="flex flex-col w-0 flex-1 overflow-hidden">
          <main class="flex flex-col flex-1 relative z-0 overflow-y-auto focus:outline-none">
            <MobileHeader />

             <slot name="page_title">
               <PageTitle page_title={{ @page_title }} :if={{ @page_title }} />
             </slot>

             <slot name="default" />

           </main>
        </div>
      </div>
      """
  end
end
```

and the `PageTitle` component:
```elixir
defmodule MyAppWeb.Admin.Layout.PageTitle do
  use MyAppWeb, [:surface_component, [slot: "page_title"]]

  @doc "The main page title"
  prop page_title, :string, required: true

  @doc "A value that is pushed on the right of the title"
  slot actions

  def render(assigns) do
    ~H"""
      <div class="border-b border-gray-200 px-4 py-4 flex items-center justify-between sm:px-6 lg:px-8">
        <div class="flex-1 min-w-0 py-2">
          <h1 class="text-lg font-medium leading-6 text-gray-900 sm:truncate">
            {{ @page_title }}
          </h1>
        </div>
        <div class="flex" :if={{ slot_assigned?(:actions) }}>
          <slot name="actions"/>
        </div>
      </div>
    """
  end
end

```

I use it like this:

Simple page with simple title:
```elixir
<Layout page_title="dashboard">
  ...
</Layout>
```

Complexe page with additional actions at the right of the page title:

```elixir
<Layout>
  <PageTitle page_title="List Vehicles">
    <template slot="actions">
      <LiveRedirect to={{ ... }}>
          New Vehicle
      </LiveRedirect>
    </template>
  </PageTitle>
</Layout>
```
The later does not work with the current implementation. I modified the code to allow "extra" slots and it seems to work but maybe I am missing something.

```suggestion
      templates: %{default: default} = templates
```

and lines 428 to 430
```diff
-              templates: %{
-                default: default
-              },
+             templates: templates,
```

With that small change, I am even able to create a sub slotable component:
```elixir
defmodule MyAppWeb.Admin.Layout.PageTitle.Actions do
  use MyAppWeb, [:surface_component, [slot: "actions"]]

  slot default

  def render(assigns) do
    ~H"""
      <span class="order-0 sm:order-1 sm:ml-3 shadow-sm rounded-md">
        <slot name="default" />
      </span>
    """
  end
end
```
And use it in my code like this:
```elixir
<Layout>
  <PageTitle page_title="List Vehicles">
    <PageTitle.Actions>
      <LiveRedirect to={{ ... }}}>
        New Vehicles
      </LiveRedirect>
    </PageTitle.Actions>
  </PageTitle>
</Layout>
```

This opens up the field of possibilities to create a powerful layout system, doesn't it?

@msaraiva This is the kind of "layout" system I had in mind in #212

Reply:
Good catch, I'll make those changes. Thanks for taking a look!

Reply:
@Malian I like the `Layout` idea 👍